### PR TITLE
Be more protective when managing duplicated imported files

### PR DESCRIPTION
**Pull request by stsewd**
When there is a duplicated file,
we are skipping the md5 and commit check/update

Maybe a fix for https://github.com/rtfd/readthedocs.org/issues/5668

We should check the db and see if we have duplicated objects in those projects ^

### CODE REVIEW

File: readthedocs/projects/tasks.py [hunk 1]
```diff
@@ -1430,25 +1430,35 @@ def _manage_imported_files(version, path, commit):
             else:
                 model_class = ImportedFile
 
-            dirpath = os.path.join(
```
Comment (stsewd) [inline readthedocs/projects/tasks.py]: dirpath, is actually the `full_path` but relative to the project's root.

File: readthedocs/projects/tasks.py [hunk 2]
```diff
@@ -1430,25 +1430,35 @@ def _manage_imported_files(version, path, commit):
...
+                    **obj_attributes,
                 )
             except model_class.MultipleObjectsReturned:
-                log.warning('Error creating ImportedFile')
-                continue
+                log.warning('Fixing duplicated objects')
+                duplicateds = (
+                    model_class.objects
+                    .filter(**obj_attributes)
+                    .order_by('-modified_date')[1:]
```
Comment (stsewd) [inline readthedocs/projects/tasks.py]: I'm removing all duplicated objects but the last one

Comment (ericholscher) [reply readthedocs/projects/tasks.py]: Seems like we shouldn't need to do this. We should do it in a migration, and write the code so it doesn't happen, no?

Comment (stsewd) [reply readthedocs/projects/tasks.py]: Yeah, that is better. But I'll need to search where we are duplicating these objects (if this is still happening), I think there was a PR that changed this, so maybe they are residuals? But I faced this locally somehow, need to investigate more.

**Comment by stsewd** [comment]
I'm going to hold on this one, first I'd like to identify all the cases where the index gets out of date with the models from the db. And maybe just delete everything from ES and the models in each build.

**Pull request closed without merging**